src/index.spec.ts

```diff
@@ -0,0 +1,301 @@
+import fetchMock from "jest-fetch-mock";
+import { mockRandomForEach } from "jest-mock-random";
+
+import { Fetch, TestResultBundle } from "@openinsights/openinsights";
+import { Provider } from "./";
+import * as template from "./lib/templateResource";
+
+import configFixture from "./fixtures/config";
+import clientInfoFixture from "./fixtures/clientInfo";
+import entryFixture from "./fixtures/entry";
+
+const settingsFixture = configFixture.settings;
+const taskFixture = configFixture.tasks[1];
+
+describe("FastlyProvider", (): void => {
+  let providerFixture: Provider;
+
+  beforeEach((): void => {
+    providerFixture = new Provider(settingsFixture) as Provider;
+    providerFixture.setSessionConfig(configFixture);
+    fetchMock.resetMocks();
+  });
+
+  describe("shouldRun", (): void => {
+    it("should always run", () => {
+      const result = new Provider(settingsFixture);
```

Comment:
you already instantiate a `Provider` in the `beforeEach()` method. Why instantiate a new one here?